core/tests/performance_tests/string_classifier_performance_test.py

```diff
@@ -0,0 +1,143 @@
+
+def benchmark(func):
+    """ A decorator to measure how much time spent by func
+
+    Args:
+        func: function. The function to be measured.
+
+    Returns:
+        function. The decorated function by this decorator.
+    """
+    def time_taken(obj, num):
+        """ Measures the time spent by func.
+
+        Args:
+            obj: object. The class instance func belongs to.
+            num: int. The number of samples to run with.
+
+        Returns:
+            Result of func.
+        """
+        start = time.time()
+        result = func(obj, num)
+        end = time.time()
+        print '%s spent %f seconds for %d instances' % (func.__name__,
+                                                        end - start, num)
+        return result
+    return time_taken
+
+class StringClassifierBenchmarker(object):
+    """ Benchmark for string classifier.
+
+    Usage:
+        benchmark = StringClassifierBenchmark()
+        benchmark.benchmark_on_train()
+        benchmark.benchmark_on_predict()
+
+    Attributes:
+        examples: list of two-element lists. An item represents a sample
+            by [doc, label].
+        predict_docs: list. A list of docs for predicting.
+        classifier: dict. The model generated by function train.
+    """
+    def __init__(self):
+        """ Initializes StringClassifierBenchmark by parsing the data
+        file containing samples
+        """
+        yaml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                 '../data/string_classifier_test.yaml')
+        doc_to_label = {}
+        with open(yaml_path, 'r') as yaml_file:
+            yaml_dict = yaml.load(yaml_file)
+            interactions = yaml_dict['states']['Home']['interaction']
+            for interaction in interactions['answer_groups'][1:]:
+                label = interaction['outcome']['feedback'][0]
+                for rule in interaction['rule_specs']:
+                    if "inputs" in rule and 'training_data' in rule['inputs']:
+                        for doc in rule['inputs']['training_data']:
+                            if doc not in doc_to_label:
+                                doc_to_label[doc] = []
+                            doc_to_label[doc].append(label)
+        self.examples = [[doc, doc_to_label[doc]] for doc in doc_to_label]
+        self.predict_docs = [doc[0] for doc in self.examples]
+        self.classifier = None
+
+    @benchmark
+    def train(self, num):
+        """ Trains a model with num samples.
+
+        Args:
+            num: int. The number of samples to run with.
+
+        Returns:
+            dict. The resulting dict representing the model.
+        """
+        string_classifier = StringClassifier()
+        string_classifier.load_examples(self.examples[:num])
+        classifier_dict = string_classifier.to_dict()
+        return classifier_dict
+
+    def benchmark_on_train(self):
```

Comment:
I looked up what "benchmark" means, and it means to evaluate or check (something) by comparison with a standard. Here, if I understand correctly, you seem to be generating the standards -- I don't really see a comparison happening here. So maybe call this generate_training_benchmarks instead?